workers/unity/Packages/com.improbable.gdk.core/Systems/WorldCommandsSendSystem.cs

```diff
@@ -75,67 +57,81 @@ protected override void OnUpdate()
                 return;
             }
 
-            for (var i = 0; i < createEntitySenderData.Length; i++)
-            {
-                var sender = createEntitySenderData.Senders[i];
-                var entity = createEntitySenderData.Entities[i];
-                foreach (var req in sender.RequestsToSend)
-                {
-                    var reqId = connection.SendCreateEntityRequest(req.Entity, req.EntityId, req.TimeoutMillis);
-                    createEntityStorage.CommandRequestsInFlight.Add(reqId.Id,
-                        new CommandRequestStore<WorldCommands.CreateEntity.Request>(entity, req, req.Context, req.RequestId));
-                }
+            var entityType = GetArchetypeChunkEntityType();
 
-                sender.RequestsToSend.Clear();
-            }
+            var createEntityType = GetArchetypeChunkComponentType<WorldCommands.CreateEntity.CommandSender>();
+            var deleteEntityType = GetArchetypeChunkComponentType<WorldCommands.DeleteEntity.CommandSender>();
+            var reserveEntityIdsType = GetArchetypeChunkComponentType<WorldCommands.ReserveEntityIds.CommandSender>();
+            var entityQueryType = GetArchetypeChunkComponentType<WorldCommands.EntityQuery.CommandSender>();
 
-            for (var i = 0; i < deleteEntitySenderData.Length; i++)
+            var chunkArray = group.CreateArchetypeChunkArray(Allocator.Temp);
```

Comment:
Would a using block be safer here? so `Dispose` happens even if something breaks? (To avoid crashing)

Reply:
IMO it would be less safe. Using will create a copy, so with unsafe code that you didn't write it's something to avoid.